packages/tables/src/Concerns/CanSelectRecords.php

```diff
@@ -45,7 +45,16 @@ public function getAllTableRecordsCount(): int
             return $this->records->total();
         }
 
-        return $this->getFilteredTableQuery()->count();
+        $query = $this->getFilteredTableQuery();
+
+        if (is_callable($this->isRecordSelectable())) {
```

Comment:
Not sure what I like better, thoughts?

```suggestion
        if ($this->isRecordSelectable() !== null) {
```